File: android/src/main/java/com/pichillilorenzo/flutter_inappwebview/types/PluginScript.java
package com.pichillilorenzo.flutter_inappwebview.types;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class PluginScript extends UserScript {
  private boolean requiredInAllContentWorlds;

  public PluginScript(@Nullable String groupName, @NonNull String source, @NonNull UserScriptInjectionTime injectionTime, @Nullable ContentWorld contentWorld, boolean requiredInAllContentWorlds) {
    super(groupName, source, injectionTime, contentWorld);
    this.requiredInAllContentWorlds = requiredInAllContentWorlds;
  }

  public boolean isRequiredInAllContentWorlds() {
    return requiredInAllContentWorlds;
  }

  public void setRequiredInAllContentWorlds(boolean requiredInAllContentWorlds) {
    this.requiredInAllContentWorlds = requiredInAllContentWorlds;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    if (!super.equals(o)) return false;

    PluginScript that = (PluginScript) o;

    return requiredInAllContentWorlds == that.requiredInAllContentWorlds;
  }

  @Override
  public int hashCode() {
    int result = super.hashCode();
    result = 31 * result + (requiredInAllContentWorlds ? 1 : 0);
    return result;
  }

  @Override
  public String toString() {
    return "PluginScript{" +
            "requiredInContentWorld=" + requiredInAllContentWorlds +
            "} " + super.toString();
  }
}

File: flutter_inappwebview_linux/linux/types/plugin_script.cc
#include "plugin_script.h"

namespace flutter_inappwebview_plugin {

PluginScript::PluginScript(const std::string& groupName, const std::string& source,
                           UserScriptInjectionTime injectionTime, bool forMainFrameOnly,
                           const std::optional<std::vector<std::string>>& allowedOriginRules,
                           std::shared_ptr<ContentWorld> contentWorld,
                           bool requiredInAllContentWorlds,
                           const std::vector<std::string>& messageHandlerNames)
    : UserScript(groupName, source, injectionTime, forMainFrameOnly, allowedOriginRules,
                 contentWorld),
      requiredInAllContentWorlds(requiredInAllContentWorlds),
      messageHandlerNames(messageHandlerNames) {}

bool PluginScript::operator==(const PluginScript& other) const {
  return UserScript::operator==(other) &&
         requiredInAllContentWorlds == other.requiredInAllContentWorlds &&
         messageHandlerNames == other.messageHandlerNames;
}

}  // namespace flutter_inappwebview_plugin
